daemon/src/surface.rs

```diff
@@ -207,12 +210,38 @@ impl Surface {
                     bail!("Directory {path:?} is empty");
                 }
 
-                let img_path = files[rand::random::<usize>() % files.len()].clone();
+                // Reset our index if we'd be out of bounds
+                if self.idx >= files.len() {
+                    self.idx = 0
+                };
+
+                // This is a bit unweildy. We're sorting every time thru this loop. But since we're
+                // walking the directory each time, I guess it's not that bad. Sorting can't be slower
+                // than reading paths off the disk, right?
+                let img_path = match self.wallpaper_info.sorting {
+                    Sorting::Random => {
+                        let index = rand::random::<usize>() % files.len();
+                        files.into_iter().nth(index).unwrap()
+                      }
+                    Sorting::Natural => {
+                        let mut sorted = files;
+                        sorted.sort();
```

Comment:
Instead of calling `sort` here, we can call [`sort_by_file_name`](https://docs.rs/walkdir/latest/walkdir/struct.WalkDir.html#method.sort_by_file_name) on Walkdir. While Walkdir says that the order is unspecified, a little test on my system shows that the returned array is already ordered; it shouldn't be a problem to order it even for `random` sorting.

Reply:
That's even better, let Walkdir do the work for us. :raised_hands: